src/call.cpp

```diff
@@ -2739,6 +2758,19 @@ bool call::process_incoming(char * msg, struct sockaddr_storage *src)
         return false;
     }
 
+    extract_transaction (txn, msg);
+
+    /* If this request came from a different peer than the one who created this
+     * dialog, store off the socket, so that when we send responses on this
+     * transaction we can send them over this socket rather than the default
+     * call_socket.
+     */
+    if (sock && (sock != call_socket) && (txn_sockets[txn] == NULL)) {
```

Comment:
I don't think so - I think for a given transaction we'l only want to communicate with one peer IP address.

If we overwrote the socket here I think the logic would have to be more complicated - e.g. we'd have to call close() on the socket when we overwrote it - and I don't think that's necessary.